components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,303 @@
+    }
+
+    @Test
+    public void testUpdateUser() throws Exception {
+    }
+
+    @DataProvider(name = "data")
+    public Object[][] data() {
+
+        String CoreClaimUri1;
+        String testMappedAttributesCore1;
+
+        String CoreClaimUri2;
+        String testMappedAttributesCore2;
+
+        String UserClaimUri1;
+        String testMappedAttributesUser1;
+
+        String UserClaimUri2;
+        String testMappedAttributesUser2;
+
+        Claim claim1 = new Claim();
+        Claim claim2 = new Claim();
+        Claim claim3 = new Claim();
+        Claim claim4 = new Claim();
+
+        CoreClaimUri1 =  "testCoreClaimURI1";
+        claim1.setClaimUri(CoreClaimUri1);
+        testMappedAttributesCore1 = "MappedAttributesCore1";
+
+        CoreClaimUri2 =  "testCoreClaimURI2";
+        claim2.setClaimUri(CoreClaimUri2);
+        testMappedAttributesCore2 = "MappedAttributesCore2";
+
+        UserClaimUri1 =  "testUserClaimURI1";
+        claim3.setClaimUri(UserClaimUri1);
+        testMappedAttributesUser1 = "MappedAttributesUser1";
+
+        UserClaimUri2 =  "testUserClaimURI2";
+        claim4.setClaimUri(UserClaimUri2);
+        testMappedAttributesUser2 = "MappedAttributesUser2";
+
+        ClaimMapping cMap1 = new ClaimMapping(claim1, testMappedAttributesCore1);
+        ClaimMapping cMap2 = new ClaimMapping(claim2, testMappedAttributesCore2);
+        ClaimMapping cMap3 = new ClaimMapping(claim3, testMappedAttributesUser1);
+        ClaimMapping cMap4 = new ClaimMapping(claim4, testMappedAttributesUser2);
+
+        ClaimMapping[] coreClaims = new ClaimMapping[]{cMap1, cMap2, cMap3, cMap4};
+
+        HashMap<String,Boolean> requiredAttributes = new HashMap<String, Boolean>() {
+            {
+                put("test1.test",true);
+            }
+        };
+
+        String[] roles = new String[]{"role1","role2","role3"};
+
+        return new Object[][]{
+                {coreClaims,requiredAttributes,roles}
+        };
+    }
+
+    @Test(dataProvider = "data")
+    public void testGetMe(Object[] cMap, HashMap<String,Boolean> required, String[] userRoles) throws Exception {
+
+        when(mockedClaimManager.getAllClaimMappings(anyString())).thenReturn((ClaimMapping[]) cMap);
+
+        SCIMUserSchemaExtensionBuilder sb = spy(new SCIMUserSchemaExtensionBuilder());
+
+        mockStatic(SCIMUserSchemaExtensionBuilder.class);
+
+        when(SCIMUserSchemaExtensionBuilder.getInstance()).thenReturn(sb);
+        when(sb.getExtensionSchema()).thenReturn(mockedScimAttributeSchema);
+
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn("testPrimaryDomain");
+
+        when(mockedUserStoreManager.getSecondaryUserStoreManager(anyString())).thenReturn(mockedUserStoreManager);
+
+        when(mockedUserStoreManager.isSCIMEnabled()).thenReturn(true);
+
+        when(mockedUserStoreManager.getRoleListOfUser(anyString())).thenReturn(userRoles);
```

Comment:
don't leave new lines after every line. leave new lines only to divide logical sections.